test/unit/ard/geographic/test_geomorphology.py

```diff
@@ -0,0 +1,306 @@
+        self.geomorphology = ard.geographic.GeomorphologyGridData()
+
+    def test_check_valid(self, subtests):
+
+        # create a mesh and try to upload it
+        y_data, x_data = np.meshgrid([-1.0, 0.0, 1.0], [0.0, 2.0])
+        z_data = np.ones_like(x_data)
+        material_data = np.array(
+            [["soil", "rock"], ["rock", "soil"], ["rock", "soil"]]
+        ).T
+
+        # set up a geomorphology grid data object
+        self.geomorphology = ard.geographic.GeomorphologyGridData()
+
+        for idx_case in range(4):
+
+            # do a setup that should fail because of check_valid
+            with subtests.test(f"check_valid bad build assertion test {idx_case}"):
+                with pytest.raises(AssertionError):
+                    self.geomorphology.set_values(
+                        x_data_in=x_data if idx_case != 0 else x_data[:1, :],
+                        y_data_in=y_data if idx_case != 1 else y_data[:1, :],
+                        z_data_in=z_data if idx_case != 2 else z_data[:1, :],
+                        material_data_in=(
+                            material_data if idx_case != 3 else material_data[:1, :]
+                        ),
+                    )
+
+            # reset to a legitimate setup
+            self.geomorphology.set_values(
+                x_data_in=x_data,
+                y_data_in=y_data,
+                z_data_in=z_data,
+                material_data_in=material_data,
+            )
+
+            # override one of the values to be invalid
+            if idx_case == 0:
+                self.geomorphology.x_data = self.geomorphology.x_data[:1, :]
+            elif idx_case == 1:
+                self.geomorphology.y_data = self.geomorphology.y_data[:1, :]
+            elif idx_case == 2:
+                self.geomorphology.z_data = self.geomorphology.z_data[:1, :]
+            else:
+                self.geomorphology.material_data = self.geomorphology.material_data[
+                    :1, :
+                ]
+
+            # make sure check valid raises an exception
+            with subtests.test(f"check_valid bad override assertion test {idx_case}"):
+                with pytest.raises(AssertionError):
+                    assert self.geomorphology.check_valid()
+
+    def test_set_values(self, subtests):
+
+        # create a mesh and try to upload it
+        y_data, x_data = np.meshgrid([-1.0, 0.0, 1.0], [0.0, 2.0])
+        z_data = np.ones_like(x_data)
+
+        # set up a geomorphology grid data object
+        self.geomorphology = ard.geographic.GeomorphologyGridData()
+        # set the values
+        self.geomorphology.set_values(
+            x_data_in=x_data,
+            y_data_in=y_data,
+            z_data_in=z_data,
+        )
+
+        # make sure the values are set in correctly
+        with subtests.test(f"set_values data equivalence tests"):
+            assert np.allclose(self.geomorphology.x_data, x_data)
```

Comment:
These assertions should be broken up into their own subtests so each subtest has only a single assertion